include/xtensor-python/pyarray.hpp

```diff
@@ -75,21 +75,188 @@ namespace pybind11
 namespace xt
 {
 
+    /**************************
+     * pybackstrides_iterator *
+     **************************/
+
+    template <class A>
+    class pybackstrides_iterator
+    {
+    public:
+
+        using self_type = pybackstrides_iterator<A>;
+
+        using value_type = typename A::size_type;
+        using pointer = const value_type*;
+        using reference = value_type;
+        using difference_type = std::ptrdiff_t;
+        using iterator_category = std::random_access_iterator_tag;
+
+        inline pybackstrides_iterator(const A* a, std::size_t offset)
+            : p_a(a), m_offset(offset)
+        {
+        }
+
+        inline reference operator*() const
+        {
+            value_type sh = p_a->shape()[m_offset];
+            value_type res = sh == 1 ? 0 : (sh - 1) * p_a->strides()[m_offset];
+            return res;
```

Comment:
I think taking a pointer on the backstrides object instead of the pyarray would allow to make this method call `pyarray_backstrides::operator[]` and would avoid code duplication.